gramps_webapi/dbmanager.py

```diff
@@ -78,10 +78,12 @@ def break_lock(self) -> None:
         if os.path.exists(os.path.join(self.path, DBLOCKFN)):
             os.unlink(os.path.join(self.path, DBLOCKFN))
 
-    def get_db(self, lock: bool = False, force_unlock: bool = False) -> DbState:
+    def get_db(self, readonly: bool = True, force_unlock: bool = False) -> DbState:
         """Open the database and return a dbstate instance.
 
-        If `lock` is `False`, will not write a lock file (use with care!).
+        If `readonly` is `True` (default), write operations will fail (note,
+        this is not enforced by Gramps but must be taken care of in Web
```

Comment:
What do you mean by enforced here?
Surely a write action on a read-only database will raise some kind of error?

Reply:
What I mean is that `self.readonly` is a flag on `DbGeneric` (the base class for database implementations) that is set to `True` when the database file is found to be read-only; it is also always true on a `ProxyDbBase` (such as the `PrivateProxyDb` we use for users who don't have `ViewPrivate` permissions). However, setting `self.readonly = True` on a `DbGeneric` child does not mean that write operations will fail. (Actually, for children of `DBAPI`, deleting objects will silently fail as the attribute is checked in the `_do_remove` method.)

I'm not saying that this is a problem in Gramps - the thing is just that when we use `get_db_handle(readonly=True)`, it should be made clear that this sets the `readonly` flag, but that is not enough to prevent write operations -  this is why the flag is explicitly checked and an error is raised in the `add_object` and `update_object` functions.

Reply:
Thanks for the explanation. Sadly, I'm still a little confused about the database layer and its components, but that is fine.

I just wanted to make sure that you don't have to check manually the read-only flag on every write action in the codebase, but rather it is checked in an underlying shared method which centralized this check. Otherwise, it sounds error-prone.